docs/API.md

```diff
@@ -1349,6 +1349,25 @@ Trying to add a batch when wrong *username* or *api key* is used:
 }
 ```
 
+Trying to add a batch with an empty list of domains:
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "error": {
+    "data": [
+      {
+        "message": "Not enough items: 0/1.",
```

Comment:
I suggest a nicer message to the human:
```
"message": "List of domains must not be empty.",
```

Reply:
This would mean overwrite the module output for this message. So far I haven't found a nice way to do it. The solution I currently have results in some tweaking that heavily relies on the module internals. And I don't feel confident in using such solution because of possible failures due to a module internal update.

Reply:
Then, would the following work?
```
"message": "List of domains must not be empty: 0/1",
```

Reply:
The approach I took is based on the [EXPERIMENTAL hack](https://metacpan.org/pod/JSON::Validator::Error#message) from the documentation:

```diff
diff --git a/lib/Zonemaster/Backend/RPCAPI.pm b/lib/Zonemaster/Backend/RPCAPI.pm
index ea6cf46..89f82b3 100644
--- a/lib/Zonemaster/Backend/RPCAPI.pm
+++ b/lib/Zonemaster/Backend/RPCAPI.pm
@@ -40,6 +40,11 @@ our %json_schemas;
 our %extra_validators;
 my $recursor = Zonemaster::Engine::Recursor->new;
 
+# overwrite the 'minItems' error message
+$JSON::Validator::Error::MESSAGES = {
+    array => {minItems => N__ 'At least %4 items expected but received %3'},
+};
+
 sub joi {
     return JSON::Validator::Joi->new;
 }
```
If you have a better approach that is not documented as experimental we can try to go with it. Otherwise I'm afraid I don't have a solution to properly set the error message for now.